src/test/architecture/test_architecture.py

```diff
@@ -312,3 +313,549 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
+        renamed_architecture_text = """
+named_paths:
+- path_name: target_path_0
+  node_chain:
+  - node_name: /node_0
+    publish_topic_name: /topic_0
+  - node_name: /node_1
+    subscribe_topic_name: /topic_0
+- path_name: target_path_1
+  node_chain:
+  - node_name: /node_1
+    publish_topic_name: /topic_1
+  - node_name: /node_0
+    subscribe_topic_name: /topic_1
+executors:
+- executor_type: single_threaded_executor
+  executor_name: executor_0
+  callback_group_names:
+  - /callback_group_0
+- executor_type: single_threaded_executor
+  executor_name: changed_executor
+  callback_group_names:
+  - /callback_group_1
+nodes:
+- node_name: /node_0
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_1
+    callback_names:
+    - /callback_0
+    - /callback_1
+  callbacks:
+  - callback_name: /callback_0
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_1
+    callback_type: subscription_callback
+    topic_name: /topic_1
+    symbol: sub
+  publishes:
+  - topic_name: /topic_0
+    callback_names:
+    - /callback_0
+  subscribes:
+  - topic_name: /topic_1
+    callback_name: /callback_1
+- node_name: /node_1
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_0
+    callback_names:
+    - /callback_2
+    - /callback_3
+  callbacks:
+  - callback_name: /callback_2
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_3
+    callback_type: subscription_callback
+    topic_name: /topic_0
+    symbol: sub
+  publishes:
+  - topic_name: /topic_1
+    callback_names:
+    - /callback_2
+  subscribes:
+  - topic_name: /topic_0
+    callback_name: /callback_3"""
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=self.renaming_architecture_text))
+        arch = Architecture('yaml', 'architecture.yaml')
+        assert arch.get_executor('executor_0')
+        assert arch.get_executor('executor_1')
+        with pytest.raises(ItemNotFoundError) as e:
+            assert arch.get_node('/changed_executor')
+        assert e
+
+        arch.rename_executor('executor_1', 'changed_executor')
+        assert arch.get_executor('executor_0')
+        assert arch.get_executor('changed_executor')
+        with pytest.raises(ItemNotFoundError) as e:
+            assert arch.get_node('/executor_1')
+        assert e
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=renamed_architecture_text))
+        arch_expected = Architecture('yaml', 'architecture.yaml')
+
+        assert set(arch.nodes) == set(arch_expected.nodes)
+        assert set(arch.communications) == set(arch_expected.communications)
+        assert set(arch.paths) == set(arch_expected.paths)
+        assert set(arch.executors) == set(arch_expected.executors)
+
+    def test_rename_topic(self, mocker):
```

Comment:
The behavior of the test_rename_topic is unstable.
This problem must be resolved before merging.

Reply:
Fix it in [eb80aa6](https://github.com/tier4/CARET_analyze/pull/156/commits/eb80aa6e100c67eec1a389bae0de3e6bf1e0661a)